spec/outputs/s3_spec.rb

```diff
@@ -229,4 +229,30 @@
       end
     end
   end
+
+  context "remove empty dirs" do
+
+    before do
+      allow(File).to receive(:directory?).with('/path/to/bar-empty').and_return(true)
+      allow(File).to receive(:directory?).with('/path/to/foo/empty-dir').and_return(true)
+      allow(File).to receive(:directory?).with('/path/to/foo-file').and_return(true)
+      allow(File).to receive(:directory?).with('/path/to/foo-file/file.tmp').and_return(false)
```

Comment:
Missing `path/to/foo` from mocked case
```suggestion
      allow(File).to receive(:directory?).with('/path/to/foo-file/file.tmp').and_return(false)
      allow(File).to receive(:directory?).with('/path/to/foo').and_return(true)
```